codec/reflectcodec/type_codec.go

```diff
@@ -159,6 +159,10 @@ func (c *genericCodec) size(
 			return 0, false, err
 		}
 
+		if size == 0 {
```

Comment:
Do we also want to enforce this on `reflect.Array`?

Reply:
or on `reflect.Struct` for serialized fields?

Reply:
on another note, would it make more sense to check this after iterating over each `innerItem`?

Reply:
Originally, this implementation banned zero length structs and zero length arrays (as everything else has a non-zero size). However, we currently use zero length structs in production: https://github.com/ava-labs/avalanchego/blob/master/vms/platformvm/txs/codec.go#L115.

So, rather than preventing 0 length values. We prevent 0 length entries in dynamically allocated structures. (i.e. anything with a length encoded in the format). This ensures that processing is O(n) in the number of bytes (whereas before it could have been O(2^n).

As for if the size should be checked in each `innerItem`, we don't need to do this. If the size is 0 of an instance, it is guaranteed to always be 0 based on the type (because we don't have optional fields).